### PR TITLE
python311Packages.pixel-ring: init at 0.1.0

### PR DESCRIPTION
**Pull request by mweinelt**
RGB LED library for ReSpeaker 4 Mic Array, ReSpeaker V2 & ReSpeaker USB 6+1 Mic Array.

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/pixel-ring/default.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+    wheel
+  ];
+
+  propagatedBuildInputs = [
+    pyusb
+    spidev
+  ];
+
+  doCheck = false; # no tests
+
```
Comment (natsukium) [inline pkgs/development/python-modules/pixel-ring/default.nix:33]: I'd be happy if you would add `pythonImportsCheck`.

File: pkgs/development/python-modules/pixel-ring/default.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  meta = with lib; {
+    description = "RGB LED library for ReSpeaker 4 Mic Array, ReSpeaker V2 & ReSpeaker USB 6+1 Mic Array";
+    homepage = "https://github.com/respeaker/pixel_ring/tree/master";
+    license = licenses.gpl2;
```
Comment (natsukium) [inline pkgs/development/python-modules/pixel-ring/default.nix:37]:
```suggestion
    license = licenses.gpl2Only;
```
I guess.

File: pkgs/development/python-modules/pixel-ring/default.nix (hunk 1 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pixel-ring/default.nix:33]: Requires SPI access on import. Let me make this more explicit.

File: pkgs/development/python-modules/pixel-ring/default.nix (hunk 2 above)
Comment (mweinelt) [reply pkgs/development/python-modules/pixel-ring/default.nix]: Applied, thanks!

**Review by natsukium — APPROVED** [review]
LGTM

**Pull request merged**